ansible_base/lib/middleware/logging/log_request.py

```diff
@@ -0,0 +1,23 @@
+import logging
+
+from ansible_base.lib.logging import thread_local
+
+logger = logging.getLogger(__name__)
+
+
+class LogRequestMiddleware:
+    """
+    Inject the request into the thread local so that it can be accessed by the logging filter.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+        thread_local.request = request
+        return response
```

Comment:
I don't get it. Isn't this what you'd want?

```python
    def __call__(self, request):
        thread_local.request = request
        response = self.get_response(request)
        thread_local.request = None
        return response
```

I don't see how `.request` is set at the _beginning_ of the request-response cycle.

I also think setting `= None` at the _end_ of the cycle is good practice, so that garbage collection can do what it needs to do.

Reply:
I think you're right, I think what I was seeing in testing was the id from the previous request.